qt/src/SourceManager.hh

```diff
@@ -48,6 +48,9 @@ struct Source {
 
 	//Additional info from original file
 	QString author, title, creator, producer, keywords, subject;
+
+	//Pdf version
+	int pdfVersionMajor, pdfVersionMinor;
```

Comment:
You should default-initialize these to zero or whatever, and check against this when using the values later on.

Reply:
Or perhaps even -1.

Reply:
Fixed.

Reply:
But i case with pdf i shouldn't check it because Poppler will return me version of pdf in these variables

Reply:
But you cannot be sure that the currently selected source is a PDF file.

Reply:
Argh. I could check it with previous switch...

Reply:
Ok, i will add check for this

Reply:
Well you can leave the static_cast, just add one additional condition if(major <= 0 or major >= 2) { version = 1.7; }

Reply:
Fixed.